### DISCUSSION TITLE
How to know that the rendering of the Chart is finished?

### DISCUSSION THREAD

**Discussion by NazTechs** [GSI-CS-CO/chart-fx#568] [General]
![image](https://user-images.githubusercontent.com/24310606/214350721-ed39af2d-12ec-42d3-9da0-96fa66a11898.png)

I want to render over 2 million points and depending on the noise level rendering all of them takes between 3 and 10 seconds. I need a solution to detect whether rendering is done or rendering in progress to show it to the user

**Comment by NazTechs** [comment]
Zoomer Plugin histogram problem
![2023-01-25_19-15-03](https://user-images.githubusercontent.com/24310606/214609265-1f1f59a9-1727-4b62-83f2-cccf2a66019e.gif)

**Comment by ennerf** [comment]
The rendering blocks the JavaFX thread, so I don't think you can provide updating indicators without moving the rendering to a background thread. You could try using AWT's Graphics2D rendering routines to draw into a PixelBuffer, and then draw the resulting image on the chart Canvas. [Foojay ](https://foojay.io/today/high-performance-rendering-in-javafx/) has more information on that.

That being said, this [data viewer](https://youtu.be/TWKAvgkoyws?t=556) shows ~5M datapoints per chart and renders fairly smoothly. Maybe you can reduce the workload a bit by e.g. doing more aggressive reductions and not rendering error bars and points. Points would also be simpler to render than lines.
